packages/api/lib/granules.js

```diff
@@ -63,10 +83,137 @@ function getGranuleProductVolume(granuleFiles = []) {
     .reduce((x, y) => x + y, 0);
 }
 
+const renameProperty = (from, to, obj) => {
+  const newObj = { ...obj, [to]: obj[from] };
+  delete newObj[from];
+  return newObj;
+};
+
+// TODO -- Docstring
+async function moveGranuleFilesAndUpdateDatastore(params) {
```

Comment:
This is a port/refactor of the original move logic. 

Originally the api model was 

1) Moving files 
2) Updating CMR metadata 
3) Updating the Dynamo granules database file object by overwriting the file object.   It was *not* preserving some of the meta keys (e.g. processing time, etc). 

The update uses the underlying @cumulus/ingest `MoveFileParams`/user 'destinations' input code to generate a list of file move objects (parallel to the original code, but per-file), then: 



1) For each file object, the postgres db is updated and files are moved -- Call moveGranuleFile which attempts to upsert over the original granule_id with new key/bucket/name values while retaining the other file metadata in the record.    If there isn't already a postgres record matching a granule_id that ties to a collection with the same name/version in the postgres database, it will *not* attempt a postgres upsert.

If the file move fails or the postgres upsert fails, the file should not have been moved and/or the database will not be updated and  the original file update information is added to a list of file model updates for postgres. 

If if the move succeeds, the updated file information is added to a list of file model updates for postgres. 

2) Update the Dynamodb granule via the api model `.update` method 

-- Once that list is assembled, the dynamoDb database write is done with a granule.file JSON that matches the original write setup.     Attempting to update that behavior such that the JSON is *updated* per file seems outside the scope of the ticket, presuming the metadata preservation doesn't create migration grief/confusion.   **Discussion is warranted before proceeding**. 

3) Update CMR metadata files

If there are errors in the move, the API returns with an error describing the state.   Regardless, the datastores should always be up-to-date with the state of the granule.files, absent a failure to write to dynamo, and in either case is an improvement from the current state of affairs.